fastscanner/services/indicators/lib/fundamental.py

```diff
@@ -107,3 +107,61 @@ async def extend_realtime(self, symbol: str, new_row: pd.Series) -> pd.Series:
 
         new_row[self.column_name()] = self._last_days[symbol]
         return new_row
+
+
+class MarketCapIndicator:
+    def __init__(self) -> None:
+        self._last_date: dict[str, date] = {}
+        self._last_market_cap: dict[str, float] = {}
+
+    @classmethod
+    def type(cls):
+        return "market_cap"
+
+    def column_name(self):
+        return self.type()
+
+    def lookback_days(self) -> int:
+        return 0
+
+    async def extend(self, symbol: str, df: pd.DataFrame) -> pd.DataFrame:
+        fundamentals = await ApplicationRegistry.fundamentals.get(symbol)
+        assert isinstance(df.index, pd.DatetimeIndex)
+
+        historical_market_cap = fundamentals.historical_market_cap
+
+        if historical_market_cap is None or historical_market_cap.empty:
+            df[self.column_name()] = np.nan
+            return df
+
+        unique_dates = np.unique(df.index.date)
+
+        date_to_market_cap = pd.Series(
+            np.nan, index=unique_dates, dtype=float, name=self.column_name()
+        )
+
+        for date in unique_dates:
+            available_dates = historical_market_cap.index[
```

Comment:
Instead of filtering the index, filter the whole series, and retrieve the first value, you wont need lines 149 and 150.